### PR TITLE
Add Exercise08

### PR DESCRIPTION
**Pull request by huynhngocminh2511**

### CODE REVIEW

File: Exercise08/Program.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Exercise08
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var n = (Console.ReadLine().Any()) ? int.Parse(Console.ReadLine()) : 0;
```
Comment (voquanghoa) [inline Exercise08/Program.cs]: To much bad code, please refactor

Comment (huynhngocminh2511) [reply Exercise08/Program.cs]: Sorry. Done

File: Exercise08/Program.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Exercise08
+{
+    class Program
+    {
+        static void Main(string[] args)
```
Comment (voquanghoa) [inline Exercise08/Program.cs:11]: Need refactor

Comment (huynhngocminh2511) [reply Exercise08/Program.cs:11]: Done

File: Exercise08/Program.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            var numbers1 = inputNumbers(new[] { 1, 2, 3, 4, 5, 6 }).ToList();
+            var numbers2 = inputNumbers(new[] { 4, 5, 6, 7, 8, 9 }).ToList();
```
Comment (voquanghoa) [inline Exercise08/Program.cs]: Remove input from user

File: Exercise08/Program.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+            var listCommon = numbers1.Join(
+                numbers2,
+                x => x,
+                y => y,
+                (x, y) => new
```
Comment (voquanghoa) [inline Exercise08/Program.cs]: `(x,y) => x`

File: Exercise08/Program.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                    {
+                        x = x
+                    })
+                .ToList();
+
+            listCommon.ForEach(Console.WriteLine);
+            Console.WriteLine();
+
+            var listOfNumberOnlyExistListFirst = numbers1
+                .Where(x => !numbers2
```
Comment (voquanghoa) [inline Exercise08/Program.cs]: Replace `!...Any(...)` to `...All(...)`

File: Exercise08/Program.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    .Any(y => y == x))
+                .ToList();
+
+            listOfNumberOnlyExistListFirst.ForEach(Console.WriteLine);
+            Console.WriteLine();
+
+            var listOfNumberExistOnlyList = numbers1
+                .Union(numbers2)
+                .Where(x => !listCommon
```
Comment (voquanghoa) [inline Exercise08/Program.cs]: Replace `!...Any(...)` to `...All(...)`

File: Exercise08/Program.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+            listOfNumberExistOnlyList.ForEach(Console.WriteLine);
+            Console.WriteLine();
+
+
+            var sum = numbers1.Select((x, i) => new
+                {
+                    x = x,
+                    i = i
+                })
+                .TakeWhile(x => true)
```
Comment (voquanghoa) [inline Exercise08/Program.cs]: `.TakeWhile(x => true)` and ` .TakeWhile(y => true)` is redundance. Try to simple it
Here is some suggestions
```C#
var count1 = numbers1.Count;
var count2 = numbers2.Count;

var total2 = Enumerable.Range(0, count1 * count2)
    .Where(x => x / count1 != x % count2)
    .Sum(x => numbers1[x / count1] * numbers2[x % count2]);

var total3 = numbers1.Sum() * numbers2.Sum() -
            Enumerable.Range(0, Math.Min(count1, count2))
                .Sum(x => numbers1[x] * numbers2[x]);

var total4 = Enumerable.Range(0, count1)
    .Sum(x => Enumerable.Range(0, count2).Where(y => x != y).Sum(y=>numbers1[x] * numbers2[y]));
```

File: Exercise08/Program.cs (hunk 3 above)
Comment (huynhngocminh2511) [reply Exercise08/Program.cs]: Done

File: Exercise08/Program.cs (hunk 4 above)
Comment (huynhngocminh2511) [reply Exercise08/Program.cs]: Done

File: Exercise08/Program.cs (hunk 5 above)
Comment (huynhngocminh2511) [reply Exercise08/Program.cs]: Done

File: Exercise08/Program.cs (hunk 6 above)
Comment (huynhngocminh2511) [reply Exercise08/Program.cs]: Done

File: Exercise08/Program.cs (hunk 7 above)
Comment (huynhngocminh2511) [reply Exercise08/Program.cs]: Done

**Pull request merged**